module_3/gift_certificates_system/repository/src/main/java/com/epam/esm/exception/EntityExistingException.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.esm.exception;
```

Comment:
From my point of view, this package should belong to the service layer since this these are "business" exceptions

Reply:
Thanks, `EntityExistingException` and `EntityNonExistentException` moved to `service` layers.
Class `IncorrectArgumentException` stay in this layer, because this exception is used in abstract class `FindAllDao<T>` in method `protected void validatePaginationOrElseThrow(EsmPagination pagination, Class<T> entity)`.

Reply:
Well, validation of the pagination better to move to the service layer, since this is validation of input parameters from controller, the dao layer should not care of that

Reply:
I thought about that too. Class `EsmPagination` is used in `dao`, because need query in the database for get maximum number of pages and compared this value (get from database) with the value entered by the user.
How then is it better to do it?